relay-server/src/actors/global_config.rs

```diff
@@ -148,8 +149,39 @@ pub struct GlobalConfigService {
 impl GlobalConfigService {
     /// Creates a new [`GlobalConfigService`].
     pub fn new(config: Arc<Config>, upstream: Addr<UpstreamRelay>) -> Self {
-        let (global_config_watch, _) = watch::channel(Arc::default());
         let (internal_tx, internal_rx) = mpsc::channel(1);
+
+        let (global_config_watch, _) = match (
+            config.relay_mode(),
+            config.has_credentials(),
+            config.global_config(),
+        ) {
+            (RelayMode::Proxy | RelayMode::Static, true, None) | (RelayMode::Managed, true, _) => {
+                relay_log::info!("global config service starting");
+                // This request will trigger the request intervals when internal_rx receives the
+                // result from upstream.
+                Self::request_global_config(upstream.clone(), internal_tx.clone());
```

Comment:
We should not make requests during initialization, as this can block relay startup. We should only do requests once the service is actually started, see my comment below in the `spawn_handler`.

Reply:
Even though the request is non-blocking ? 

also, is it fine to then initialize the watch a default global config even if we don't want to use that?  
 
or we should initialize with an option<arc<config>>, where it will be none in the time between initialization and the calling of the spawn_handler